pkg/operator/ceph/config/livenessprobe.go

```diff
@@ -62,3 +51,27 @@ func ConfigureLivenessProbe(daemon rookv1.KeyType, container v1.Container, healt
 
 	return container
 }
+
+func GetLivenessProbeWithDefaults(desiredProbe, defaultProbe *v1.Probe) *v1.Probe {
+	// No need to allow overriding of the liveness probe command
+	desiredProbe.Handler = defaultProbe.Handler
+
+	// If the user has not specified thresholds and timeouts, set them to the same values as
+	// in the default liveness probe created by Rook.
+	if desiredProbe.FailureThreshold == 0 {
+		desiredProbe.FailureThreshold = defaultProbe.FailureThreshold
+	}
+	if desiredProbe.PeriodSeconds == 0 {
+		desiredProbe.PeriodSeconds = defaultProbe.PeriodSeconds
+	}
+	if desiredProbe.SuccessThreshold == 0 {
+		desiredProbe.SuccessThreshold = defaultProbe.SuccessThreshold
+	}
+	if desiredProbe.TimeoutSeconds == 0 {
+		desiredProbe.TimeoutSeconds = defaultProbe.TimeoutSeconds
```

Comment:
In the default probe we only set `InitialDelaySeconds` see [here](https://github.com/rook/rook/blob/master/pkg/operator/ceph/controller/spec.go#L542) or [here](https://github.com/rook/rook/blob/master/pkg/operator/ceph/object/spec.go#L280). So, anyway, the above fields will be 0?

Reply:
If we weren't setting it before, this should still give us the same before as before, right? So K8s must use its own defaults if set to 0.

Reply:
Correct.